### PR TITLE
Empty activity page

### PR DESCRIPTION
**Pull request by jeroach**
US105774: https://rally1.rallydev.com/#/238990806412d/detail/userstory/299142688352

### CODE REVIEW

File: components/d2l-quick-eval/d2l-quick-eval.js [hunk 1]
```diff
@@ -35,12 +36,13 @@ class D2LQuickEval extends mixinBehaviors(
 					width: 250px;
 					margin-left: .25rem;
 				}
-				.d2l-quick-eval-top-bar {
+				d2l-quick-eval-view-toggle {
+					clear: both;
+					float: left;
```
Comment (CarolBusman) [inline components/d2l-quick-eval/d2l-quick-eval.js]: `clear` followed by `float` will put the toggle on a new line. If we only `float: left;`, the toggle will be on the same line as the title.

File: components/d2l-quick-eval/d2l-quick-eval.js [hunk 2]
```diff
@@ -42,7 +42,10 @@ class D2LQuickEval extends mixinBehaviors(
 				}
 				.d2l-quick-eval-header {
 					float: left;
-					padding-top: 0.25rem;
+				}
+				.d2l-quick-eval-header-with-toggle {
```
Comment (jeroach) [inline components/d2l-quick-eval/d2l-quick-eval.js]: This ensures proper spacing between the header and either the toggle or the submissions view.

It also ensures that:
- if the toggle is hidden, the search and filter are inline with the header text
- if the toggle is visible, the search and filter are inline with the toggle

**Comment by jeroach** [comment]
Tests all pass locally - single test failure in travis is an activity card test:
https://travis-ci.com/BrightspaceHypermediaComponents/activities/builds/112069161

Kicking again

File: components/d2l-quick-eval/d2l-quick-eval-activities.js [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+					clear: both;
+				}
+				d2l-hm-search {
+					display: inline-block;
+					width: 250px;
+					margin-left: .25rem;
+				}
+			</style>
+			<div class="d2l-quick-eval-activity-list-modifiers">
+				<d2l-hm-filter
```
Comment (jeroach) [inline components/d2l-quick-eval/d2l-quick-eval-activities.js]: I just dropped filter and search in here for demo purposes and to ensure alignment.

The wire up (and creating of the variables, etc.) will happen under the 2 other stories for adding filter and search to activities view - one of which I just picked up :)

File: components/d2l-quick-eval/d2l-quick-eval-view-toggle.js [hunk 4]
```diff
@@ -96,10 +96,14 @@ class D2LQuickEvalViewToggle extends QuickEvalLocalize(PolymerElement) {
 		};
 	}
 	_selectSubmissions() {
-		this.currentSelected = this._viewTypes.submissions;
+		if (this.currentSelected !== this._viewTypes.submissions) {
```
Comment (jeroach) [inline components/d2l-quick-eval/d2l-quick-eval-view-toggle.js]: Now that the toggle is again storing it's own state, no need to change logic here as it was earlier.

Adding these checks here to be explicit (as to not fire the event if we're clicking on the already selected button). Carol and I discussed offline that these checks might not be strictly needed, but we decided to leave them in to clearly indicate our intent and purpose.

File: components/d2l-quick-eval/d2l-quick-eval.js [hunk 5]
```diff
@@ -58,19 +63,19 @@ class D2LQuickEval extends mixinBehaviors(
...
-					<h1>[[headerText]]</h1>
-				</template>
-				<div class="flex-break" hidden$="[[!activitiesViewEnabled]]"></div>
-				<d2l-quick-eval-view-toggle hidden$="[[!activitiesViewEnabled]]"></d2l-quick-eval-view-toggle>
+			<template is="dom-if" if="[[headerText]]">
+				<h1 class="d2l-quick-eval-header-with-toggle" hidden$="[[!activitiesViewEnabled]]"">[[headerText]]</h1>
+				<h1 class="d2l-quick-eval-header" hidden$="[[activitiesViewEnabled]]"">[[headerText]]</h1>
+			</template>
+			<d2l-quick-eval-view-toggle current-selected="submissions" hidden$="[[!activitiesViewEnabled]]"></d2l-quick-eval-view-toggle>
+			<div class="submissionsView" hidden$="[[_showActivitiesView]]">
```
Comment (jeroach) [inline components/d2l-quick-eval/d2l-quick-eval.js]: Eventually, this div will be replaced with a `d2l-quick-eval-submissions` tag. Likely when this lines up with @AlexBedley 's refactor :)

File: components/d2l-quick-eval/d2l-quick-eval.js [hunk 6]
```diff
@@ -45,7 +45,7 @@ class D2LQuickEval extends mixinBehaviors(
 				}
 				.d2l-quick-eval-header-with-toggle {
 					float: left;
-					padding-bottom: 1rem;
+					padding-bottom: 1.8rem;
```
Comment (jeroach) [inline components/d2l-quick-eval/d2l-quick-eval.js]: Based on: https://d2l.slack.com/messages/CBTK3JB6D/

Comment (jeroach) [reply components/d2l-quick-eval/d2l-quick-eval.js]: Hmmm, it actually looks like the `36px` aka `1.8rem` is to the bottom of the "Eval", not to the bottom of the container, which is lower due to the tail of the Q.

I'll check with Dan.

Comment (jeroach) [reply components/d2l-quick-eval/d2l-quick-eval.js]: Checked with Dan - he's confirming the spacing. I'll update once it's confirmed.

File: components/d2l-quick-eval/d2l-quick-eval-activities.js [hunk 7]
```diff
@@ -0,0 +1,46 @@
+import {html, PolymerElement} from '@polymer/polymer/polymer-element.js';
```
Comment (jeroach) [inline components/d2l-quick-eval/d2l-quick-eval-activities.js]: Alex and I chatted offline and went with Alex's suggestion to clean this file up to the bare minimum.

Diff is cleaner now, and we can have nice small diffs as we add each additional feature.

File: components/d2l-quick-eval/d2l-quick-eval.js [hunk 8]
```diff
@@ -321,6 +334,14 @@ class D2LQuickEval extends mixinBehaviors(
 			this._moreSearchResults = false;
 		}
 	}
+
+	_toggleView(e) {
+		if (e.detail.view === 'submissions') {
```
Comment (AlexBedley) [inline components/d2l-quick-eval/d2l-quick-eval.js]: I'm not a huge fan of the magic strings we use everywhere like this one but that's probably outside the scope of this PR.

File: demo/d2l-quick-eval/d2l-quick-eval-activities.html [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+			import '@polymer/iron-demo-helpers/demo-snippet';
+			import 'd2l-typography/d2l-typography.js';
+			import './update-direction.js';
+			import startMockServer from './interceptor';
+			import mappingsParser from './demo-data-parser';
+			import data from './table';
+
+			const mappings = mappingsParser(data);
+			// Parameters: data, network logging, api request delay
+			startMockServer(mappings, true, 100);
```
Comment (AlexBedley) [inline demo/d2l-quick-eval/d2l-quick-eval-activities.html]: Just like the component, I would rip out all this unused stuff

Comment (jeroach) [reply demo/d2l-quick-eval/d2l-quick-eval-activities.html]: Fixed

File: components/d2l-quick-eval/d2l-quick-eval.js (hunk 8 above)
Comment (jeroach) [reply components/d2l-quick-eval/d2l-quick-eval.js]: Yeah I agree - but out of scope

**Pull request merged**